subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpErrorStatusCodeException.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.resource.transport.http;
+
+import org.gradle.internal.exceptions.Contextual;
+
+/**
+ * Signals that HTTP response has been received successfully but an error code is encountered (neither 2xx/3xx nor 404).
```

Comment:
:x: "neither 2xx/3xx nor 404" - That's incorrect based on the logic in `HttpClientHelper.wasSuccessful`.

`statusCode >= 200 && statusCode < 400`

Reply:
We check "wasMissing" before, so this is correct.

Reply:
Thanks for pointing out. The 404 check is here: https://github.com/gradle/gradle/blob/5c75c8c9be85181db1145c810b33b7ead54562a1/subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientHelper.java#L152